test/sql/timestamp.sql

```diff
@@ -731,6 +740,29 @@ SELECT  to_char(d, 'YYYY-MM-DD') AS d,
 FROM generate_series('2015-01-01' :: date, '2020-12-01', '6 month') AS ts,
      unnest(array[ts :: date]) AS d;
 
+-- Test timezones support with different bucket sizes
+BEGIN;
+-- Timestamptz type is displayed in the session timezone.
+-- To get consistent results during the test we temporary set the session
+-- timezone to the known one.
+SET TIME ZONE '+00';
+
+-- Moscow is UTC+3 in the year 2021. Let's say you are dealing with '1 day' bucket.
+-- In order to calculate the beginning of the bucket you have to take LOCAL
+-- Moscow time and throw away the time. You will get the midnight. The new day
+-- starts 3 hours EARLIER in Moscow than in UTC+0 time zone, thus resulting
+-- timestamp will be 3 hours LESS than for UTC+0. You might need a pencil and
+-- a piece of paper to ensure that it's true and the results are correct.
+SELECT bs, tz, to_char(ts_out, 'YYYY-MM-DD HH24:MI:SS TZ') as res
+FROM unnest(array['Europe/Moscow', 'UTC']) as tz,
+     unnest(array['12 hours', '1 day', '1 month', '4 months', '1 year']) as bs,
+     unnest(array['2021-07-12 12:34:56 Europe/Moscow' :: timestamptz]) as ts_in,
+     unnest(array[timescaledb_experimental.time_bucket_ng(bs :: interval, ts_in, timezone => tz)]) as ts_out
+ORDER BY tz, bs :: interval;
+
+-- Restore previously used time zone.
+ROLLBACK;
```

Comment:
For context: I remember that we stumbled over situations where "ephemeral tables" were treated differently from named tables. In this case it does not make a difference, but this might be something that we should consider for more complicated usage.